opencog/guile/SchemeEval.cc

```diff
@@ -813,6 +813,90 @@ void * SchemeEval::c_wrap_eval_h(void * p)
 	return self;
 }
 
+/**
+ * Evaluate a string containing a scheme expression, returning a HandleSeq.
+ * If an evaluation error occurs, then the error is logged to the log
+ * file, and an empty HandleSeq is returned.  If the result of evaluation
+ * is not a list, no error is logged, but an empty HandleSeq is still
+ * returned. If any item in the list is not a Handle, still no error is
+ * returned, but an empty HandleSeq is still returned. Otherwise, if the
+ * result of evaluation is a list of Handle, a HandleSeq is returned.
+ */
+HandleSeq SchemeEval::eval_q(const std::string& expr)
```

Comment:
I really really really don't like this ... can't you just return a ListLink with all the handles in it?  That requires no new C code.